src/apis/impl/MessageGatewayApiImpl.ts

```diff
@@ -21,66 +35,98 @@ export class MessageGatewayApiImpl implements MessageGatewayApi {
     this.specId = specId;
     this.authHeader = (!!username && !!password) ? new Header(
       "Authorization",
-      "Basic "+Buffer.from(`${username}:${password}`).toString('base64')
+      "Basic " + Buffer.from(`${username}:${password}`).toString('base64')
     ) : null;
-    this.headers = [new Header("Content-Type", "application/json")]
+    this.headers = [new Header("Content-Type", "application/json")];
+    this.errorHandler = errorHandler;
+    this.sanitizer = sanitizer;
   }
 
-  private readonly fetchImpl?: (input: RequestInfo, init?: RequestInit) => Promise<Response>;
-  private readonly msgGtwUrl: string;
-  private readonly specId: string;
-  private readonly authHeader: XHR.Header | null;
-  private readonly headers: Header[];
+  async sendEmail(recipientEmail: string, email: EmailMessage): Promise<boolean> {
+    if (!this.authHeader) return false;
+
+    const emailTo = this.sanitizer.validateEmail(recipientEmail)
+
+    if (!emailTo) {
+      throw this.errorHandler.createErrorWithMessage(`Invalid email address ${recipientEmail}`)
+    }
 
-  async sendEmail(recipientEmail: string, email: EmailMessage): Promise<XHR.Data | null> {
-    if (!this.authHeader) return null;
     const res = await XHR.sendCommand('POST',
-      `${this.msgGtwUrl}/${this.specId}/email/to/${recipientEmail}`,
+      `${this.msgGtwUrl}/${this.specId}/email/to/${emailTo}`,
       this.headers.concat([this.authHeader]),
       email,
       this.fetchImpl
-    );
+    ).catch(e => {
+      throw this.errorHandler.createErrorFromAny(e)
+    });
 
-    if (res.statusCode < 400) return res;
-    else throw new Error(`Message Gateway returned error response ${res.statusCode}`);
+    if (res.statusCode < 400) {
+      return true;
+    } else {
+      throw this.errorHandler.createErrorWithMessage(`Message Gateway returned error response ${res.statusCode}`)
```

Comment:
Users won't know what is the MessageGateway, it is not exposed to them. 
Tell something like `iCure could not send an email to ${emailTo}, due to error ${res.statusCode}`
But for me, this message should even disappear because it will be treat in createErrorFromAny above